src/actions/posts.js

```diff
@@ -364,6 +364,40 @@ export function editPost(post) {
     });
 }
 
+export function setUnreadPost(userId, postId) {
+    return async (dispatch, getState) => {
+        dispatch({type: ChannelTypes.POST_UNREAD_REQUEST}, getState);
+
+        let unreadChan;
+        const state = getState();
```

Comment:
This call should be moved to happen after the network request. Depending on how long the network request takes, the state might have changed, so we could be looking at old data by doing it this way.

Reply:
It's intentional, by having it there I'll get the amount it had when requested which feels as close to the expected outcome as I can think of. having it after the network request might mean that an action taken after this one, might affect it. Same thing can happen with a previous action being slower with this one.

So it seems that we might have some race conditions wherever I put this, what I'm not sure is which one is the one that will feel more ackward or is it going to happen more often.